src/dataset/generic_dataset.py

```diff
@@ -1,8 +1,515 @@
+        inp = self._get_input(img, trans_input)
+        ret = {'image': inp}
+        gt_det = {'bboxes': [], 'scores': [], 'clses': [], 'cts': []}
+
+        pre_cts, track_ids = None, None
+        if self.opt.tracking:
+            pre_image, pre_anns, frame_dist = self._load_pre_data(video_idx, frame_idx)
+            if flipped:
+                pre_image = pre_image[:, ::-1, :].copy()
+                pre_anns = self._flip_anns(pre_anns, width)
+            if self.opt.same_aug_pre and frame_dist != 0:
+                trans_input_pre = trans_input
+                trans_output_pre = trans_output
+            else:
+                c_pre, aug_s_pre, _ = self._get_aug_param(c, s, width, height, disturb=True)
+                s_pre = s * aug_s_pre
+                trans_input_pre = dataset_utils.get_affine_transform(c_pre, s_pre, rot, [self.opt.input_res.width,
+                                                                                         self.opt.input_res.height])
+                trans_output_pre = dataset_utils.get_affine_transform(c_pre, s_pre, rot, [self.opt.output_res.width,
+                                                                                          self.opt.output_res.height])
+            pre_img = self._get_input(pre_image, trans_input_pre)
+            pre_hm, pre_cts, track_ids = self._get_pre_dets(pre_anns, trans_input_pre, trans_output_pre)
+            ret['pre_img'] = pre_img
+            if self.opt.pre_hm:
+                ret['pre_hm'] = pre_hm
+
+        self._init_ret(ret, gt_det)
+        calib = self._get_calib(img_info, width, height)
+
+        num_objs = min(len(anns), self.max_objs)
+        for k in range(num_objs):
+            ann = anns[k]
+            cls_id = int(ann['category_id'])
+            if cls_id > self.num_classes or cls_id <= -999:
+                continue
+            bbox, bbox_amodal = self._get_bbox_output(ann['bbox'], trans_output, height, width)
+            if cls_id <= 0 or ('iscrowd' in ann and ann['iscrowd'] > 0):
+                self._mask_ignore_or_crowd(ret, cls_id, bbox)
+                continue
+            self._add_instance(
+                    ret, gt_det, k, cls_id, bbox, bbox_amodal, ann,
+                    trans_output, aug_s,
+                    calib, pre_cts, track_ids)
+
+        return ret
+
+    def _get_specific_nuscenes_frame_debug(self, idx):
+        if idx == 0:
+            filename = 'samples/CAM_BACK/n008-2018-08-28-13-40-50-0400__CAM_BACK__1535478715187558.jpg'
+        else:
+            filename = 'samples/CAM_BACK/n008-2018-08-21-11-53-44-0400__CAM_BACK__1534867443937696.jpg'
+        frame_abs = 0
+        for vid in self.videos:
+            for frame in vid:
+                sample_data = self.nusc.get('sample_data', frame)
+                if sample_data['filename'] == filename:
+                    return frame_abs
+
+                frame_abs += 1
+
+        raise Exception
```

Comment:
It's better to be as specific as possible with exceptions. This here could be a `ValueError`. Also, it would be good to have an informative message.